crates/vm/levm/src/call_frame.rs

```diff
@@ -18,6 +20,41 @@ pub struct Log {
     pub data: Bytes,
 }
 
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+pub struct Stack {
+    pub stack: Vec<U256>,
+}
+
+impl Stack {
+    pub fn pop(&mut self) -> Result<U256, VMError> {
+        self.stack.pop().ok_or(VMError::StackUnderflow)
+    }
+
+    pub fn push(&mut self, value: U256) -> Result<(), VMError> {
+        if self.stack.len() >= STACK_LIMIT {
+            return Err(VMError::StackOverflow);
+        }
+        self.stack.push(value);
+        Ok(())
+    }
+
+    pub fn len(&self) -> usize {
+        self.stack.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.stack.is_empty()
+    }
+
+    pub fn get(&self, index: usize) -> Result<&U256, VMError> {
+        self.stack.get(index).ok_or(VMError::StackUnderflow)
```

Comment:
A `StackUnderflow ` error occurs when an item is attempted to be retrieved from a stack, but the stack doesn't have enough elements, I think for now it's fine to represent it like this. We could have another error for example like `StackIndexOutOfBounds` but I believe it's kind of the same.

Reply:
Isn't `StackUnderflow` when we try to pop an item from the stack and it is empty? For example if the stack has 10 elements and we want to get the 20th element then I don't think that `StackUnderflow` is an appropriate name for that scenario. It would be accurate in a `pop` though, not necessarily in a `get`. Your suggestion, `StackIndexOutOfBounds`, is more precise I'd say.